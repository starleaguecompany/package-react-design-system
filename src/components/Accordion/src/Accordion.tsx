import * as React from 'react'
import { useStyles, useMultiValue, safeInvoke, isNull } from '@starleaguecompany/package-react-utils'

import { Space } from '../../Space'

import { AccordionProps } from '../types/Accordion.types'
import styles from '../styles/Accordion.module.less'

export * from '../types/Accordion.types'

/**
 * @description A content area which can be collapsed and expanded
 *
 * @component
 * @example
 * ```jsx
 * <Accordion activeKeys={['credit']}>
 *   <Accordion.Item key="credit" title="О кредите">content</Accordion.Item>
 * </Accordion>
 * ```
 */
const Accordion = React.forwardRef<HTMLDivElement, AccordionProps>((props, ref) => {
  const { defaultIndex, separator, background, scrollIntoView, children, className, onChange, ...restProps } = props
  const [innerValue, setInnerValue] = useMultiValue<number>(defaultIndex || [])

  const cx = useStyles(styles)

  const classNames = cx(className, {
    [`separator-${separator}`]: separator,
    [`background-${background}`]: background,
  })

  const handleCollapse = React.useCallback(
    event => {
      const value = Number(event.currentTarget.dataset.index)

      setInnerValue(value, val => {
        safeInvoke(onChange, val)
      })
    },
    [innerValue]
  )

  const separatorSpace = separator === 'space' ? 16 : 0

  const panels = React.useMemo(() => {
    return React.Children.map(children as React.ReactElement, (child, index) => {
      if (isNull(child)) return

      const childProps = {
        ...child.props,
        id: index,
        active: innerValue.indexOf(index) >= 0,
        scrollIntoView,
        onCollapse: handleCollapse,
      }

      return React.cloneElement(child, childProps)
    })
  }, [innerValue, children])

  return (
    <Space
      ref={ref}
      data-qa="Accordion"
      size={separatorSpace}
      direction="vertical"
      className={classNames}
      {...restProps}
    >
      {panels}
    </Space>
  )
})

Accordion.defaultProps = {
  defaultIndex: [],
  separator: 'space',
  scrollIntoView: true,
}

export default Accordion
