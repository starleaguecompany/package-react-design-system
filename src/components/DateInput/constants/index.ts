export const DAYS_OF_WEEK = ['Пн', 'Вт', 'Ср', 'Чт', 'Пт', 'Сб', 'Вс']
export const MONTHS = [
  'Январь',
  'Февраль',
  'Март',
  'Апрель',
  'Май',
  'Июнь',
  'Июль',
  'Август',
  'Сентябрь',
  'Октябрь',
  'Ноябрь',
  'Декабрь',
]
