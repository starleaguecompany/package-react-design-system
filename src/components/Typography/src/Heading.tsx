import * as React from 'react'
import { useStyles } from '@starleaguecompany/package-react-utils'

import { HeadingProps } from '../types/Heading.types'
import styles from '../styles/Heading.module.less'

/**
 * @description The Heading component is used for all headings.
 *
 * @component
 * @example
 * ```jsx
 * <Heading level={2}>
 *   Heading
 * </Heading>
 * ```
 */
const Heading = React.forwardRef<HTMLHeadingElement, HeadingProps>((props, ref) => {
  const { as, level, className, children, ...restProps } = props
  const cx = useStyles(styles)

  const classNames = cx(className, 'container', {
    [`size-${level}`]: level,
  })

  return React.createElement(
    as || (`h${level}` as React.ElementType),
    {
      ref: ref,
      className: classNames,
      ...restProps,
    },
    children
  )
})

Heading.defaultProps = {
  level: 1,
}

export { Heading }
