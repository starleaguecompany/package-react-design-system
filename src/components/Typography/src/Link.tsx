import * as React from 'react';
import { useStyles } from '@starleaguecompany/package-react-utils';

import { LinkProps } from '../types/Link.types';
import styles from '../styles/Link.module.less';

/**
 * @description The Link component is used for anchor links. This component renders a a element by default
 *
 * @component
 * @example
 * ```jsx
 * <Link href="#">Link</Link>
 * ```
 */
const Link = React.forwardRef<HTMLAnchorElement, LinkProps>((props, ref) => {
  const { color, strong, className, children, ...restProps } = props;
  const cx = useStyles(styles);

  const classNames = cx(className, 'container', {
    [`color-${color}`]: color,
    strong,
  });

  return (
    <a ref={ref} className={classNames} {...restProps}>
      {children}
    </a>
  );
});

Link.defaultProps = {
  color: 'blue',
};

export { Link };
