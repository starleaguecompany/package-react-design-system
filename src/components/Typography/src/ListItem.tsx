import * as React from 'react'
import { useStyles } from '@starleaguecompany/package-react-utils'

import { ListItemProps } from '../types/ListItem.types'
import styles from '../styles/List.module.less'

/**
 * @description List item element.
 *
 * @component
 * @example
 * ```jsx
 * <ListItem>...</ListItem>
 * ```
 */
const ListItem = React.forwardRef<HTMLLIElement, ListItemProps>((props, ref) => {
  const { icon, className, children, ...restProps } = props
  const cx = useStyles(styles)

  const classNames = cx(className, 'item', {
    'with-icon': icon,
  })

  return (
    <li ref={ref} className={classNames} {...restProps}>
      {icon ? <span className={cx('icon')}>{icon}</span> : null}
      {children}
    </li>
  )
})

export { ListItem }
