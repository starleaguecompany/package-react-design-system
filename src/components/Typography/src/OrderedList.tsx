import * as React from 'react';
import { useStyles } from '@starleaguecompany/package-react-utils';

import { OrderedListProps } from '../types/OrderedList.types';
import styles from '../styles/List.module.less';

/**
 * @description Ordered list element.
 *
 * @component
 * @example
 * ```jsx
 * <OrderedList>...</OrderedList>
 * ```
 */
const OrderedList = React.forwardRef<HTMLOListElement, OrderedListProps>((props, ref) => {
  const { className, children, ...restProps } = props;
  const cx = useStyles(styles);

  const classNames = cx(className, 'ordered');

  return (
    <ol ref={ref} className={classNames} {...restProps}>
      {children}
    </ol>
  );
});

export { OrderedList };
