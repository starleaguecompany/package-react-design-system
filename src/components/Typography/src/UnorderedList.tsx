import * as React from 'react';
import { useStyles } from '@starleaguecompany/package-react-utils';

import { UnorderedListProps } from '../types/UnorderedList.types';
import styles from '../styles/List.module.less';

/**
 * @description Unordered list element.
 *
 * @component
 * @example
 * ```jsx
 * <UnorderedList>...</UnorderedList>
 * ```
 */
const UnorderedList = React.forwardRef<HTMLUListElement, UnorderedListProps>((props, ref) => {
  const { className, children, ...restProps } = props;
  const cx = useStyles(styles);

  const classNames = cx(className, 'unordered');

  return (
    <ul ref={ref} className={classNames} {...restProps}>
      {children}
    </ul>
  );
});

export { UnorderedList };
