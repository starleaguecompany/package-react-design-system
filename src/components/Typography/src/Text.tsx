import * as React from 'react';
import { useStyles } from '@starleaguecompany/package-react-utils';

import { FONT_SIZES } from '../../../constants/sizes';

import { TextProps } from '../types/Text.types';
import styles from '../styles/Text.module.less';

/**
 * @description The Text component is used for single line or multiline text. The component renders a div element by default
 *
 * @component
 * @example
 * ```jsx
 * <Text as="div" size="xl" strong>Text</Text>
 * ```
 */
const Text = React.forwardRef<HTMLDivElement, TextProps>((props, ref) => {
  const {
    as,
    size,
    strong,
    uppercase,
    capitalize,
    lowercase,
    truncate,
    nowrap,
    lineThrough,
    className,
    children,
    ...restProps
  } = props;
  const cx = useStyles(styles);

  const classNames = cx(className, {
    [`size-${size}`]: true,
    strong,
    uppercase,
    capitalize,
    lowercase,
    truncate,
    nowrap,
    lineThrough,
  });

  return React.createElement(
    as as React.ElementType,
    {
      ref: ref,
      className: classNames,
      ...restProps,
    },
    children
  );
});

Text.defaultProps = {
  size: FONT_SIZES.S14,
  as: 'div',
};

export { Text };
