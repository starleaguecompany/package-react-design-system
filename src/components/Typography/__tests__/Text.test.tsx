import * as React from 'react'
import { render, screen } from '@testing-library/react'

import { FONT_SIZES } from '../../../constants/sizes'
import { Text } from '../src/Text'

describe('Text', () => {
  test('should render correctly with no props', () => {
    const { container } = render(<Text />)

    expect(container).toMatchSnapshot()
    expect(container.firstChild).toHaveClass('size-14', { exact: true })
  })

  test('should render correctly with text', () => {
    const { container } = render(<Text>Testing text</Text>)

    expect(container).toMatchSnapshot()
    expect(container.firstChild).toHaveTextContent('Testing text')
    expect(screen.getByText('Testing text')).toBeVisible()
  })

  test('should render correctly with attributes', () => {
    const { container } = render(
      <Text id="test-id" className="test-class">
        Testing text
      </Text>
    )

    expect(container).toMatchSnapshot()
    expect(screen.getByText('Testing text')).toHaveAttribute('id', 'test-id')
    expect(screen.getByText('Testing text')).toHaveClass('test-class')
  })

  test('should render correctly with paragraph element', () => {
    const { container } = render(<Text as="p">Testing text</Text>)

    expect(container).toMatchSnapshot()
    expect(container.firstChild.nodeName).toBe('P')
    expect(container.firstChild).toHaveTextContent('Testing text')
    expect(screen.getByText('Testing text')).toBeVisible()
  })

  test('should render correctly with div element', () => {
    const { container } = render(<Text as="div">Testing text</Text>)

    expect(container).toMatchSnapshot()
    expect(container.firstChild.nodeName).toBe('DIV')
    expect(container.firstChild).toHaveTextContent('Testing text')
    expect(screen.getByText('Testing text')).toBeVisible()
  })

  test('should render correctly with size prop', () => {
    const { container } = render(<Text size={FONT_SIZES.S10}>Testing text</Text>)

    expect(container).toMatchSnapshot()
    expect(screen.getByText('Testing text')).toHaveClass('size-10')
  })

  test('should render correctly with upper prop', () => {
    const { container } = render(<Text uppercase>Testing text</Text>)

    expect(container).toMatchSnapshot()
    expect(screen.getByText('Testing text')).toHaveClass('uppercase')
  })

  test('should render correctly with lower prop', () => {
    const { container } = render(<Text lowercase>Testing text</Text>)

    expect(container).toMatchSnapshot()
    expect(screen.getByText('Testing text')).toHaveClass('lowercase')
  })

  test('should render correctly with capitalize prop', () => {
    const { container } = render(<Text capitalize>Testing text</Text>)

    expect(container).toMatchSnapshot()
    expect(screen.getByText('Testing text')).toHaveClass('capitalize')
  })

  test('should render correctly with strong prop', () => {
    const { container } = render(<Text strong>Testing text</Text>)

    expect(container).toMatchSnapshot()
    expect(screen.getByText('Testing text')).toHaveClass('strong')
  })

  test('should render correctly with truncate prop', () => {
    const { container } = render(<Text truncate>Testing text</Text>)

    expect(container).toMatchSnapshot()
    expect(screen.getByText('Testing text')).toHaveClass('truncate')
  })

  test('should render correctly with lineThrough prop', () => {
    const { container } = render(<Text lineThrough>Testing text</Text>)

    expect(container).toMatchSnapshot()
    expect(screen.getByText('Testing text')).toHaveClass('lineThrough')
  })
})
