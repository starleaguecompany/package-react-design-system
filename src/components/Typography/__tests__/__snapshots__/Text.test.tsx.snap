// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should render correctly with attributes 1`] = `
<div>
  <div
    class="test-class size-14"
    id="test-id"
  >
    Testing text
  </div>
</div>
`;

exports[`Text should render correctly with capitalize prop 1`] = `
<div>
  <div
    class="size-14 capitalize"
  >
    Testing text
  </div>
</div>
`;

exports[`Text should render correctly with div element 1`] = `
<div>
  <div
    class="size-14"
  >
    Testing text
  </div>
</div>
`;

exports[`Text should render correctly with lineThrough prop 1`] = `
<div>
  <div
    class="size-14 lineThrough"
  >
    Testing text
  </div>
</div>
`;

exports[`Text should render correctly with lower prop 1`] = `
<div>
  <div
    class="size-14 lowercase"
  >
    Testing text
  </div>
</div>
`;

exports[`Text should render correctly with no props 1`] = `
<div>
  <div
    class="size-14"
  />
</div>
`;

exports[`Text should render correctly with paragraph element 1`] = `
<div>
  <p
    class="size-14"
  >
    Testing text
  </p>
</div>
`;

exports[`Text should render correctly with size prop 1`] = `
<div>
  <div
    class="size-10"
  >
    Testing text
  </div>
</div>
`;

exports[`Text should render correctly with strong prop 1`] = `
<div>
  <div
    class="size-14 strong"
  >
    Testing text
  </div>
</div>
`;

exports[`Text should render correctly with text 1`] = `
<div>
  <div
    class="size-14"
  >
    Testing text
  </div>
</div>
`;

exports[`Text should render correctly with truncate prop 1`] = `
<div>
  <div
    class="size-14 truncate"
  >
    Testing text
  </div>
</div>
`;

exports[`Text should render correctly with upper prop 1`] = `
<div>
  <div
    class="size-14 uppercase"
  >
    Testing text
  </div>
</div>
`;
